sdk/ai/azure-ai-generative/azure/ai/generative/index/_mlindex.py

```diff
@@ -241,16 +251,27 @@ def as_langchain_vectorstore(self, credential: Optional[TokenCredential] = None)
                 store = None
                 engine = self.index_config.get("engine")
                 if engine == "langchain.vectorstores.FAISS":
-                    from azure.ai.generative.index._langchain.vendor.vectorstores.faiss import FAISS
-
-                    embeddings = EmbeddingsContainer.from_metadata(self.embeddings_config.copy()).as_langchain_embeddings(credential=credential)
+                    embeddings = EmbeddingsContainer.from_metadata(
+                        self.embeddings_config.copy()
+                    ).as_langchain_embeddings(credential=credential)
 
                     fs, uri = url_to_fs(self.base_uri)
 
                     with tempfile.TemporaryDirectory() as tmpdir:
                         fs.download(f"{uri.rstrip('/')}/index.pkl", f"{str(tmpdir)}")
                         fs.download(f"{uri.rstrip('/')}/index.faiss", f"{str(tmpdir)}")
-                        store = FAISS.load_local(str(tmpdir), embeddings)
+
+                        try:
+                            from langchain.vectorstores import FAISS
+
+                            store = FAISS.load_local(str(tmpdir), embeddings)
+                        except Exception as e:
+                            logger.warning(
+                                f"Failed to load FAISS Index using installed version of langchain, retrying with vendored FAISS VectorStore.\n{e}"
+                            )
+                            from azureml.rag.langchain.vendor.vectorstores.faiss import FAISS
```

Comment:
This needs to import from azure.ai.generative: `from azure.ai.generative.index._langchain.vendor.vectorstores.faiss import FAISS`

Reply:
Good catch. I did a quick search for "azureml.rag". will fix them all in next iteration.